src/main/java/com/sonymobile/tools/gerrit/gerritevents/dto/rest/Topic.java

```diff
@@ -0,0 +1,121 @@
+/*
+ *  The MIT License
+ *
+ *  Copyright 2013 Jyrki Puttonen. All rights reserved.
+ *  Copyright 2013 Sony Mobile Communications AB. All rights reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package com.sonymobile.tools.gerrit.gerritevents.dto.rest;
+
+import com.sonymobile.tools.gerrit.gerritevents.GerritQueryException;
+import com.sonymobile.tools.gerrit.gerritevents.GerritQueryHandler;
+import com.sonymobile.tools.gerrit.gerritevents.dto.attr.Change;
+import com.sonymobile.tools.gerrit.gerritevents.dto.attr.PatchSet;
+import com.sonymobile.tools.gerrit.gerritevents.dto.events.ChangeBasedEvent;
+import net.sf.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class to represent Topic related information.
+ */
+public class Topic {
+    private static final Logger logger = LoggerFactory.getLogger(ChangeBasedEvent.class);
+
+    private final String name;
+    private Map<Change, PatchSet> changes = null;
+
+    /**
+     * Standard Constructor.
+     *
+     * @param name the topic name.
+     */
+    public Topic(String name) {
+        this.name = name;
+    }
+
+    /**
+     * @return the name of topic.
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Gets all change-patchset pairs related to this topic.
+     * After first call data is cached.
+     *
+     * @param gerritQueryHandler the query handler, responsible for the queries to gerrit.
+     * @return the map of pairs change-patchset related to this topic.
+     */
+    public Map<Change, PatchSet> getChanges(GerritQueryHandler gerritQueryHandler) {
+        if (changes == null) {
+            changes = new HashMap<Change, PatchSet>();
```

Comment:
If something fails below, you will end up with an empty change list in this topic "forever" And will not be able to retrieve them on a second try.